### ISSUE TITLE
OPA - Add helper text for complex Constraints parameters

### ISSUE THREAD

**Issue by izaac** [rancher/dashboard#670]
**What kind of request is this:**
Enhancement

**Steps to reproduce (least amount of steps as possible):**

- Add the `labels` parameter in a constraint using the `k8srequiredlabels` template with the following:

Key:
`labels`
Value:
`[ { "allowedRegex": "^$|^[a-zA-Z0-9]+", "key": "gatekeeper" } ]`

**Result:**
```
 { "type": "error", "links": {}, "code": "Invalid", "message": "K8sRequiredLabels.constraints.gatekeeper.sh \"test-labels\" is invalid: spec.parameters.labels: Invalid value: \"string\": spec.parameters.labels in body must be of type array: \"string\"", "status": 422 }
```

***Expectation***
We should add a UI information/warning element explaining this limitation in the Edit Form. And ask the user to use the Edit as YAML for complex parameters values.

**Environment information**
- Rancher version: `v2.4-head (05/19/2020)` 8e20687a4
- Installation option (single install/HA): Single

**Comment by codyrancher** [comment]
@izaac what was the exact input for parameters? I was able to add what you described, save it and view it.

![image](https://user-images.githubusercontent.com/55104481/82376334-c7103280-99d6-11ea-884b-776b4268b918.png)

**Comment by izaac** [comment]
Sure @codyrancher  I am using it like this using the form (without clicking Edit YAML):

![Screen Shot 2020-05-19 at 2 51 47 PM](https://user-images.githubusercontent.com/27659/82382524-fc218280-99e0-11ea-96b0-fcc7c8ca447d.png)

![Screen Shot 2020-05-19 at 2 35 59 PM](https://user-images.githubusercontent.com/27659/82382569-0ba0cb80-99e1-11ea-9a4e-09e25384ccc8.png)
![Screen Shot 2020-05-19 at 2 36 03 PM](https://user-images.githubusercontent.com/27659/82382589-15c2ca00-99e1-11ea-9d17-62c1efbe0a29.png)



This is the POST payload:

```json
{
    "type": "constraints.gatekeeper.sh.k8srequiredlabels",
    "spec": {
        "match": {
            "excludedNamespaces": ["cattle-system", "gatekeeper-system", "kube-public", "kube-system"],
            "kinds": [{
                "apiGroups": [""],
                "kinds": ["Namespaces"]
            }],
            "labelSelector": {
                "matchExpressions": []
            },
            "namespaceSelector": {
                "matchExpressions": []
            }
        },
        "parameters": {
            "labels": "[ { \"allowedRegex\": \"^$|^[a-zA-Z0-9]+\", \"key\": \"gatekeeper\" } ]",
            "message": "All namespaces must use the `gatekeeper`  label"
        },
        "enforcementAction": "deny"
    },
    "metadata": {
        "name": "test-labels2",
        "annotations": {
            "cattle.io/description": null
        }
    }
}
```

Response:

```json
{
    "type": "error",
    "links": {},
    "code": "Invalid",
    "message": "K8sRequiredLabels.constraints.gatekeeper.sh \"test-labels2\" is invalid: []: Invalid value: map[string]interface {}{\"kind\":\"K8sRequiredLabels\", \"metadata\":map[string]interface {}{\"annotations\":map[string]interface {}{\"cattle.io/description\":\"\"}, \"generation\":1, \"uid\":\"dff401a2-9a1a-11ea-8c49-0af40943c8e4\", \"name\":\"test-labels2\", \"creationTimestamp\":\"2020-05-19T21:51:20Z\"}, \"spec\":map[string]interface {}{\"enforcementAction\":\"deny\", \"match\":map[string]interface {}{\"kinds\":[]interface {}{map[string]interface {}{\"apiGroups\":[]interface {}{\"\"}, \"kinds\":[]interface {}{\"Namespaces\"}}}, \"labelSelector\":map[string]interface {}{\"matchExpressions\":[]interface {}{}}, \"namespaceSelector\":map[string]interface {}{\"matchExpressions\":[]interface {}{}}, \"excludedNamespaces\":[]interface {}{\"cattle-system\", \"gatekeeper-system\", \"kube-public\", \"kube-system\"}}, \"parameters\":map[string]interface {}{\"labels\":\"[ { \\\"allowedRegex\\\": \\\"^$|^[a-zA-Z0-9]+\\\", \\\"key\\\": \\\"gatekeeper\\\" } ]\", \"message\":\"All namespaces must use the `gatekeeper`  label\"}}, \"type\":\"constraints.gatekeeper.sh.k8srequiredlabels\", \"apiVersion\":\"constraints.gatekeeper.sh/v1beta1\"}: validation failure list:\nspec.parameters.labels in body must be of type array: \"string\"",
    "status": 422,
    "type": "error"
}
```

So for this it's not clear for a user new to the UI in what cases he should use Edit YAML or use the Input Fields, that's the goal of this issue.

**Comment by codyrancher** [comment]
@izaac @maggieliu are we okay with this messaging? If not, let me know what you'd like and I'll update it. I only show the banner when Edit as Form is being used.

![Screen Shot 2020-10-05 at 3 19 38 PM](https://user-images.githubusercontent.com/55104481/95138120-55831780-071e-11eb-82d3-598fb973315e.png)

**Comment by izaac** [comment]
@codyrancher this works for me, but will wait on @maggieliu preference.

**Comment by maggieliu** [comment]
@codyrancher I am good with the text. But I think this should be more of a @cloudnautique question.

**Comment by cloudnautique** [comment]
@codyrancher / @maggieliu I'm good with the text.

**Comment by codyrancher** [comment]
What if we get rid of the parameters key value form and instead just have a yaml editor with a schema derived from reading the selected template?

**Comment by codyrancher** [comment]
@izaac I think we should close this issue in favor of rancher/dashboard#1777

**Comment by izaac** [comment]
Closing in favor of #1777
